registry/consul.go

```diff
@@ -0,0 +1,205 @@
+//
+// Copyright (c) 2018
+// IOTech
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package registry
+
+import (
+	"errors"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"fmt"
+	consulapi "github.com/hashicorp/consul/api"
+)
+
+type ConsulClient struct {
```

Comment:
what is purpose of this structure?

Reply:
Sorry maybe I've asked wrong, but I mean 'ConsulClient struct'. Currently it's simple proxy with 1 field. What sense of such structure? Do you want make some additional logic in it? If not - currently it complicates things.

Reply:
These are copy from edgex internal package.
And I made a interface for consul.
It contains three functions: 1.init 2.registry 3.load consul config
https://github.com/edgexfoundry/device-sdk-go/blob/53a1df7d6021b66f9b7fd1d3468e6c2b0634e688/registry/registry.go